docs/paper/dev/guides/pdc.md

```diff
@@ -0,0 +1,148 @@
+---
+slug: /dev/pdc
+---
+
+# Persistent Data Container (PDC)
+
+The Persistent Data Container (PDC) is a way to store custom data on a whole range of objects; such as items, entities, and blocks.
+The Full list of classes that support the PDC are:
+
+- [Chunk](#chunk)
+- [World](#world)
+- [Entity](#entity)
+- [TileState](#tilestate)
+- [ItemMeta](#itemmeta)
+- [Structure](#structure)
+
+## What is it used for?
+In the past, developers resorted to storing data within the NBT tags of an object. 
+This was a very hacky way of storing data, and was prone to having conflicts between different plugins
+when they used the same keys. This is resolved by the use of `NamespacedKey`.
+Therefore, the PDC is a much more stable way of storing data, and is much easier to use.
+
+As the name implies, the PDC is persistent. This means that the data stored in the PDC will be saved to disk, 
+and will be loaded back into the object when the server is restarted. This is great for storing attributes of an object,
+such as the owner of an item, or the custom name of an entity.
+
+## Adding Data
+To store data in the PDC, there are a few things you need first. The first is a `NamespacedKey` which is used to identify the data.
+The second is a `PersistentDataContainer` which is the object you want to store the data on. The third is the data itself.
+
+```java
+// Create a NamespacedKey
+NamespacedKey key = new NamespacedKey(pluginInstance, "example-key");
+
+ItemStack item = new ItemStack(Material.DIAMOND);
+// ItemMeta implements PersistentDataHolder so we can get the PDC from it
+ItemMeta meta = item.getItemMeta();
+item.getPersistentDataContainer().set(key, PersistentDataType.STRING, "I love Tacos!");
+item.setItemMeta(meta);
+```
+
+:::note
+
+It is considered good practice to reuse the `NamespacedKey` objects. Constructing them from a static context is ideal for this.
```

Comment:
This is a bit weird. Static context is rather troublesome for namespaced keys as they require your plugin instance which is not really available in a static context unless you do some fun static state.

Paper specifically un-deprecates https://jd.papermc.io/paper/1.19/org/bukkit/NamespacedKey.html#%3Cinit%3E(java.lang.String,java.lang.String), I am unsure if we want to promote its usage however.

Reply:
personally, I've been moving towards using that constructor for various reasons

1) They're glorified strings, not instance derived magical things; purely making them instance based for the sake of injecting Plugin#gatName() as part of the glorified string is meh
2) stability, especially in terms of portability and renaming stuff; I.e. I can copy that key to another plugin and use it with 0 strife

I honestly think that that is the most appropriate constructor for creating keys in general, IMHO

Reply:
Yea it is more of a "safety" for newer developers in my mind.
If you *know what you are doing*  that constructor is great. If you don't, better to stick to the plugin constructor than a `new NamespacedKey("minecraft", "my-key");`

Reply:
I have re-written this now